### PR TITLE
[#799] Restrict security context of pods by default

### PR DESCRIPTION
**Pull request by brusdev**
The operator pod and the broker pods can run with a safer and restricted security context without issues. Restricting the security context of the pods by default allows to deploy a cluster of brokers also in Kubernetes namespaces with the restricted policy.

### CODE REVIEW

**Comment by brusdev** [comment]
I incorporated the commits from #757 in this PR to support an easy way for users to relax the security contexts.

**Comment by jbertram** [comment]
My experience with the operator is pretty limited so I won't be able to provide any meaningful feedback.

File: docs/help/operator.md [hunk 1]
```diff
@@ -1040,6 +1040,22 @@ spec:
 
 When deploying the above custom resource the operator will spread matching pods among the given topology
 
+### Container SecurityContext
+
+The ActiveMQArtemis custom resource offers a container level SecurityContext option for the broker holds security configuration that will be applied to the containers.
```
Comment (gtully) [inline docs/help/operator.md:1045]: the broker _that_ holds

Think that is missing

**Review by gtully — APPROVED** [review]
Looks great.

**Comment by brusdev** [comment]
> My experience with the operator is pretty limited so I won't be able to provide any meaningful feedback.

@jbertram sorry it was my mistake

**Comment by brusdev** [comment]
@howardgao I incorporated the commits from #743 in this PR test the operator deployment in a restricted namespace. I removed the changes related to the security contexts of the broker pods because they were causing a conflict with the commits from #757 and I split the test to be consistent.

**Review by gaohoward — APPROVED** [review]

**Comment by kevinfrommelt** [comment]
@brusdev I just upgraded to 1.1.0 and it seems like the default security context that gets added is not compatible with the `activemq-artemis-broker-init` image for tags older than `1.0.25` b/c the default user in the image is `root`.
```
Warning  Failed     39m (x12 over 41m)   kubelet            Error: container has runAsNonRoot and image has non-numeric user (root), cannot verify user is non-root (pod: "dev-1-artemis-ss-2_dev-1(012c6399-3953-4291-b969-ed3999177ba4)", container: dev-1-artemis-container-init)
```
```
» docker inspect quay.io/artemiscloud/activemq-artemis-broker-init:1.0.23 | jq '.[0].Config.User'
"root"
» docker inspect quay.io/artemiscloud/activemq-artemis-broker-init:1.0.24 | jq '.[0].Config.User'
"root"
» docker inspect quay.io/artemiscloud/activemq-artemis-broker-init:1.0.25 | jq '.[0].Config.User'
"185"
```

Aside from updating to the latest, I can work around this by adding the following:
```
apiVersion: broker.amq.io/v1beta1
kind: ActiveMQArtemis
metadata:
  name: broker
  namespace: activemq-artemis-operator
spec:
  deploymentPlan:
    containerSecurityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - ALL
      runAsNonRoot: true
      runAsUser: 185
      seccompProfile:
        type: RuntimeDefault
```

**Comment by brusdev** [comment]
@kevinfrommelt I confirm that the activemq-artemis-broker-init image for tags older than 1.0.25 have the default user `root` and they need your workaround to be used because for security reasons it is better using container as non root.

**Pull request merged**